src/hash_table.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
+    def __init__(self, key: str, val: Union[str, float], next: Node = None):
+        self.key = key
+        self.val = val
+        self.next = next
+
+
+class ChainHashTable:
+    def __init__(self, n: int):
+        self.__bucket = [Node(None, None)] * n
+
+    def insert(self, key: str, val: Union[str, float]) -> bool:
+        node: Node = self.__bucket[self._hash(key)]
+        while node.next:
+            node = node.next
+            if node.key == key:
+                node.val = val
+                return True
+        node.next = Node(key, val)
+        return True
```

Comment:
I think this function is still working. But i dont think it's the best for while node.next, if you dont process node.next in while logic, then it seems u always dont care about the last element, which have `next` is None.